x-pack/plugins/licensing/public/index.ts

```diff
@@ -5,12 +5,20 @@
  * 2.0.
  */
 
-// TODO: https://github.com/elastic/kibana/issues/110902
-/* eslint-disable @kbn/eslint/no_export_all */
+import type { PluginInitializerContext } from 'src/core/public';
 
-import { PluginInitializerContext } from 'src/core/public';
 import { LicensingPlugin } from './plugin';
 
-export * from '../common/types';
-export { LicensingPluginSetup, LicensingPluginStart } from './types';
+export type {
+  LicenseCheckState,
+  LicenseType,
```

Comment:
`LicenseType` declaration depends on the enum `LICENSE_TYPE`. Should we export the enum as well? Or is it OK with keeping it internal?

Reply:
exported `LICENSE_TYPE` type as well. turned out several plugins already use. However, I believe we should deprecate it in favor of https://github.com/elastic/kibana/blob/6157313e212a0308345805fa30233b6275311404/x-pack/plugins/licensing/common/types.ts#L158